spring-kafka/src/main/java/org/springframework/kafka/listener/RecoveringBatchErrorHandler.java

```diff
@@ -191,4 +194,24 @@ private void commit(Consumer<?, ?> consumer, MessageListenerContainer container,
 		}
 	}
 
+	private BatchListenerFailedException getBatchListenerFailedException(Throwable throwable) {
```

Comment:
@artembilan 
If Exception and Cause are the same, it can defend.
But, If it has a circular relationship, it cannot escape the recursion.

```
A.cause() -> B
B.cause() -> A
```

This is an unusual case.
Please comment if you don't need to defend.

Reply:
OK. Let it be. I wanted to avoid that local `Set` to store causes, but if you both are OK with this, let's merge!